compiler/rustc_abi/src/layout.rs

```diff
@@ -934,3 +775,287 @@ pub trait LayoutCalculator {
         })
     }
 }
+
+fn univariant(
+    this: &(impl LayoutCalculator + ?Sized),
+    dl: &TargetDataLayout,
+    fields: &IndexSlice<FieldIdx, Layout<'_>>,
+    repr: &ReprOptions,
+    kind: StructKind,
+    niche_bias_start: bool,
```

Comment:
I think the code could benefit from this being an `enum` with documentation explaining exactly what it decides